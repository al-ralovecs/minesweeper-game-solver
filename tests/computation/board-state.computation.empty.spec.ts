import { disposition } from '../fixtures/state/3x3.empty.json';
import BoardDto from "../../src/dto/board.dto";
import BoardStateComputation from "../../src/computation/board-state.computation";

describe('BoardStateComputation', () => {
    test('check how computes empty board', () => {
        const board = new BoardDto(disposition);
        const boardStateComputation = new BoardStateComputation(board.height, board.width);

        boardStateComputation.setBoard = board;
        boardStateComputation.do();

        expect(boardStateComputation.getBoardState).toMatchSnapshot();
    });
});
