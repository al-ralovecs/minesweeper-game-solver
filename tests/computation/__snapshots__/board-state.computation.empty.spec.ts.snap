// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardStateComputation check how computes empty board 1`] = `
BoardStateDto {
  "action": Array [
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
  ],
  "actionList": Array [],
  "adjFlagsOnBoard": Array [
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
  ],
  "adjUnrevealed": Array [
    Array [
      3,
      5,
      3,
    ],
    Array [
      5,
      8,
      5,
    ],
    Array [
      3,
      5,
      3,
    ],
  ],
  "board": Array [
    Array [
      -1,
      -1,
      -1,
    ],
    Array [
      -1,
      -1,
      -1,
    ],
    Array [
      -1,
      -1,
      -1,
    ],
  ],
  "flagConfirmed": Array [
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
  ],
  "flagOnBoard": Array [
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
  ],
  "height": 3,
  "livingWitnesses": Array [],
  "numOfHidden": 9,
  "revealed": Array [
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
  ],
  "totalFlags": 0,
  "totalFlagsConfirmed": 0,
  "width": 3,
}
`;
